scripts/Diffraction/isis_powder/pearl.py

```diff
@@ -78,10 +78,13 @@ def should_subtract_empty_inst(self):
     @contextmanager
     def _apply_temporary_inst_settings(self, kwargs, run):
 
-        # set temporary settings
+        # set temporary settings, Check has to occur before updating attributes,
+        # otherwise it would assumed the longmode vars are cached.
         if not self._inst_settings.long_mode == bool(kwargs.get("long_mode")):
+            self._inst_settings.update_attributes(kwargs=kwargs)
             self._switch_long_mode_inst_settings(kwargs.get("long_mode"))
-        self._inst_settings.update_attributes(kwargs=kwargs)
```

Comment:
Could you not have just moved this line before the `if` to gain the same result?

Reply:
(actually, this is what @TTitcombe already suggested, why wasn't this done?)